File: arch/all-ios/bootstrap/ui.m
#include <stdarg.h>
#include <runtime.h>

#include "appdelegate.h"

void DisplayError(char *fmt, ...)
{
    va_list ap;
    NSString *format = [NSString stringWithCString:fmt encoding:NSISOLatin1StringEncoding];
    BootstrapDelegate *ad = (BootstrapDelegate *)[[UIApplication sharedApplication] delegate];

    va_start(ap, fmt);

    NSString *text = [[NSString alloc] initWithFormat:format arguments:ap];

    [ad ShowAlert:text withTitle:@"AROS bootstrap error"];
    [text release];

    va_end(ap);
}

File: arch/all-hosted/bootstrap/ui.c
/*
    Copyright © 1995-2014, The AROS Development Team. All rights reserved.
    $Id$
*/

#include <stdarg.h>
#include <stdio.h>

void DisplayError(char *fmt, ...)
{
    va_list args;
    
    va_start(args, fmt);
    vprintf(fmt, args);
    va_end(args);

    printf("\n");
}
